File: 2_gyak/liskov/Liskov.cpp
class Madar {
	public virtual void repul() {};
};

class Program {
	public void fgv(Madar &madar) {
          madar.repul();
	}
};

class Sas : public Madar
{};

class Pingvin : public Madar
{};

int main (int argc, char **argv)
{
     Program program;
     Madar madar;
     program.fgv(madar);

     Sas sas;
     program.fgv(sas);

     Pingvin pingvin;
     program.fgv(pingvin);

File: 2_gyak/liskov/Liskov.java

class Madar {
    public void repul() {
        System.out.println("Repül.");
    }
}

class Sas extends Madar {   
}

class Pingvin extends Madar {    
}

public class Liskov {
    public void repul() {};
    
    
    public static void main(String[] args) {
        Madar madar = new Madar();
        madar.repul();
        
        Sas sas = new Sas();
        sas.repul();
        
        Pingvin pingvin = new Pingvin();
        pingvin.repul();
        
    }
    
}
